dna/plot.py

```diff
@@ -13,3 +18,48 @@ def plot_at_k_scores(model_names: pd.Series, model_scores, model_colors, plot_pa
     plt.legend(loc=0)
     plt.savefig(plot_path)
     plt.clf()
+
+
+def create_distribution_plots(agg_results: dict, output_dir: str):
+    scores_by_dataset = agg_results[0]['test_scores']['scores_by_dataset_id']
+    if len(list(scores_by_dataset.keys())) == 0:
+        print('No results, skipping')
+        return
+
+    metric_keys = scores_by_dataset[list(scores_by_dataset.keys())[0]].keys()
+    results_dict = {}
+    # aggregates metrics over all datasets
+    for metric_key in metric_keys:
+        new_metric_list = []
+        for index, dataset_name in enumerate(scores_by_dataset.keys()):
+            new_metric_list.append(scores_by_dataset[dataset_name][metric_key])
```

Comment:
scores_by_dataset.keys() is called a lot. Maybe save the result of this in a local variable

Reply:
Would totally be good